includes/core/DataTransferObjects/class-data_transfer_object.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace WCPay\Core\DataTransferObjects;
+
+use ReflectionClass;
+use ReflectionException;
+
+abstract class Data_Transfer_Object
+{
+	/**
+	 * Return array of all properties.
+	 *
+	 * @return array
+	 */
+	public function to_array() {
+		$class = new ReflectionClass($this);
+		$data = [];
+		foreach ($class->getProperties() as $reflectionProperty) {
+			$data[$reflectionProperty->getName()] = $reflectionProperty->getValue();
```

Comment:
It might be a good idea to focus on properties with specific access (public or private).

Reply:
This is base method. If something needs to be removed, the parent class could override a method and create a custom to_array function.